### ISSUE TITLE
useSeoMeta does not set values on SSR when using reactive syntax

### ISSUE THREAD

**Issue by dan-sarich** [nuxt/nuxt#23470]
### Environment

Nuxt project info:
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-21dpcd?file=app.vue

![image](https://github.com/nuxt/nuxt/assets/12992802/b88b4038-55e4-41bf-a9cf-48efb40b32f4)

### Describe the bug

- When using the reactive syntax (() =>) useSeoMeta does not set the proper meta tags on SSR. 
- By right clicking and viewing the page source, you can see the tags are not present and title is `undefined`

Current workaround -> useHead with reactive syntax (() =>) does in fact set the tags properly but is much more verbose.

With **useSeoMeta**: values undefined
<img width="633" alt="image" src="https://github.com/nuxt/nuxt/assets/12992802/bf31e6ca-d4f6-4dc2-8e82-59e9651801ef">

With **useHead**: values present
<img width="597" alt="image" src="https://github.com/nuxt/nuxt/assets/12992802/fcb5ad56-f33d-45de-a440-cc3268e78db7">

**Comment by danielroe** [comment]
cc: @harlan-zw

**Comment by harlan-zw** [comment]
Awaiting the fetch also fixes it as a workaround.

```ts
const { data } = await useFetch('/api/demo');

useSeoMeta(() => ({
  title: { textContent: data.value?.title },
  description: data.value?.description,
  ogTitle: data.value?.['og:title'],
  ogDescription: data.value?.['og:description'],
}));
```

I believe the issue is that useSeoMeta will immediately resolve the reactivity to map the keys properly, whereas useHead will resolve the reactivity as it's actually rendering the head data.

**Comment by dan-sarich** [comment]
@harlan-zw Thanks for that, For our use case we do not want to await the fetch.

**Comment by harlan-zw** [comment]
The fix for this will be available in v1.8 which will go out in the next couple of days.

**Comment by hi-reeve** [comment]
> The fix for this will be available in v1.8 which will go out in the next couple of days.

any update?
